homeassistant/components/humidifier/__init__.py

```diff
@@ -161,7 +161,7 @@ def device_class(self) -> HumidifierDeviceClass | str | None:
     def state_attributes(self) -> dict[str, Any]:
         """Return the optional state attributes."""
         supported_features = self.supported_features or 0
-        data = {}
+        data: dict[str, int | str | None] = {}
```

Comment:
We normally check for None value before including the attribute in the attributes.

Reply:
It seems it's gotten lost in some integrations though. It's currently a mixed bag.

Reply:
Besides, I do not agree with that comment; These are not extra attributes.
Additionally, this PR is not trying to change existing behavior, but type the existing behavior.

Reply:
We do this for state attributes too not only for extra attributes.

Reply:
I'm only commenting on something that we may want to adjust for the future.

Reply:
I don't agree; however, not the point of this PR either.

Reply:
What do you don't agree with? Regardless what we decide we should look over this so that it's consistent.

Reply:
Let discuss that outside of this PR as this is not related to this PR.